eventmesh-registry-plugin/eventmesh-registry-zookeeper/src/main/java/org/apache/eventmesh/registry/zookeeper/service/ZookeeperRegistryService.java

```diff
@@ -0,0 +1,311 @@
+import org.apache.eventmesh.common.config.CommonConfiguration;
+import org.apache.eventmesh.common.utils.ConfigurationContextUtil;
+import org.apache.eventmesh.registry.zookeeper.constant.ZookeeperConstant;
+import org.apache.eventmesh.registry.zookeeper.pojo.Instance;
+import org.apache.eventmesh.registry.zookeeper.util.JsonUtils;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.data.Stat;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class ZookeeperRegistryService implements RegistryService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ZookeeperRegistryService.class);
+
+    private static final AtomicBoolean INIT_STATUS = new AtomicBoolean(false);
+
+    private static final AtomicBoolean START_STATUS = new AtomicBoolean(false);
+
+    private String serverAddr;
+
+    public CuratorFramework zkClient = null;
+
+    private Map<String, EventMeshRegisterInfo> eventMeshRegisterInfoMap;
+
+    @Override
+    public void init() throws RegistryException {
+        boolean update = INIT_STATUS.compareAndSet(false, true);
+        if (!update) {
+            return;
+        }
+        eventMeshRegisterInfoMap = new HashMap<>(ConfigurationContextUtil.KEYS.size());
+        for (String key : ConfigurationContextUtil.KEYS) {
+            CommonConfiguration commonConfiguration = ConfigurationContextUtil.get(key);
+            if (null == commonConfiguration) {
+                continue;
+            }
+            if (StringUtils.isBlank(commonConfiguration.namesrvAddr)) {
+                throw new RegistryException("namesrvAddr cannot be null");
+            }
+            this.serverAddr = commonConfiguration.namesrvAddr;
+            break;
+        }
+    }
+
+    @Override
+    public void start() throws RegistryException {
+        boolean update = START_STATUS.compareAndSet(false, true);
+        if (!update) {
+            return;
+        }
+        try {
+            RetryPolicy retryPolicy = new ExponentialBackoffRetry(1000, 5);
+            zkClient = CuratorFrameworkFactory.builder()
+                .connectString(serverAddr)
+                .retryPolicy(retryPolicy)
+                .namespace(ZookeeperConstant.NAMESPACE)
+                .build();
+            zkClient.start();
+
+        } catch (Exception e) {
+            logger.error("[ZookeeperRegistryService][start] error", e);
+            throw new RegistryException(e.getMessage());
```

Comment:
You can directly throw the exception to upper layer, since you don't need to log here, this may cause the exception boom.
```suggestion
            throw new RegistryException("ZookeeperRegistry starting failed", e);
```